builtin/game/falling.lua

```diff
@@ -33,11 +61,92 @@ core.register_entity(":__builtin:falling_node", {
 				end
 			end
 		end
+		local def = core.registered_nodes[node.name]
+		if not def then
+			-- Don't allow unknown nodes to fall
+			core.log("info",
+				"Unknown falling node removed at "..
+				core.pos_to_string(self.object:get_pos()))
+			self.object:remove()
+			return
+		end
 		self.meta = meta
-		self.object:set_properties({
-			is_visible = true,
-			textures = {node.name},
-		})
+		if def.drawtype == "airlike" then
+			self.object:set_properties({
+				is_visible = false,
+			})
+		elseif def.drawtype == "torchlike" or def.drawtype == "signlike" then
+			local textures
+			if def.tiles and def.tiles[1] then
+				if def.drawtype == "torchlike" then
+					textures = { "("..def.tiles[1]..")^[transformFX", def.tiles[1] }
+				else
+					textures = { def.tiles[1] }
+				end
+			end
+			local vsize
+			if def.visual_scale then
+				vsize = { x = def.visual_scale, y = def.visual_scale, z = def.visual_scale }
+			end
+			self.object:set_properties({
+				is_visible = true,
+				visual = "upright_sprite",
+				visual_size = vsize,
+				textures = textures,
+				glow = def.light_source,
+			})
+		else
+			local itemstring = node.name
+			if core.is_colored_paramtype(def.paramtype2) then
+				itemstring = core.itemstring_with_palette(itemstring, node.param2)
+			end
+			local vsize
+			if def.visual_scale then
+				vsize = { x = def.visual_scale * SCALE, y = def.visual_scale * SCALE, z = def.visual_scale * SCALE }
+			end
+			self.object:set_properties({
+				is_visible = true,
+				wield_item = itemstring,
+				visual_size = vsize,
+				glow = def.light_source,
+			})
+		end
+		-- Rotate entity
+		if def.drawtype == "torchlike" then
+			self.object:set_yaw(math.pi*0.25)
+		elseif (node.param2 ~= 0 and (def.wield_image == ""
+				or def.wield_image == nil))
+				or def.drawtype == "signlike" then
```

Comment:
What about nodebox drawtype? If you change `default:sign_wall_wood` to a falling node, the rotation is not correct.
EDIT: Neither are meshes -> example: `default:torch_wall`, the Y rotation is always off by 180°.